src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientTest.java

```diff
@@ -158,8 +157,8 @@ public void failedWithBearerTokenFromConfiguration() throws Exception {
         try {
             message = saslServer.evaluateResponse(message);
         } catch (SaslException e) {
-            assertTrue(e.getCause() instanceof RealmUnavailableException);
-            RealmUnavailableException cause = (RealmUnavailableException) e.getCause();
+            assertTrue(e.getCause() instanceof IllegalArgumentException);
```

Comment:
The point here is an Exception is being used to control authentication flow when really it is just evidence verification that has failed which can already be represented using the API without throwing an Exception.